patient-service/src/main/java/com/raf/si/patientservice/service/impl/HealthRecordServiceImpl.java

```diff
@@ -163,4 +177,142 @@ private boolean canGetConfidential(){
                 .stream()
                 .anyMatch(permission -> permission.equals(PERMITTED_DOC));
     }
+
+    private HealthRecord getRecordByLbp(UUID lbp) {
+        // dohvati iz baze korisnika
+        Patient patient = patientService.findPatient(lbp);
+        HealthRecord healthRecord = patient.getHealthRecord();
+        if(healthRecord == null) {
+            String errMessage = String.format("pacijent sa lbp '%s' nema zdravstveni karton", lbp);
+            log.info(errMessage);
+            throw new InternalServerErrorException(errMessage);
+        }
+        return healthRecord;
+    }
+
+    @Override
+    public MessageResponse updateHealthrecord(UpdateHealthRecordRequest updateHealthRecordRequest, UUID lbp) {
+
+        // proveri da li su vrednosti koje je korisnik poslao dobre
+        BloodType bt = BloodType.forName(updateHealthRecordRequest.getBlodtype());
+        //log.info(bt.toString());
+        if(bt == null) {
+            String errMessage = String.format("Nepoznata krvna grupa '%s'", updateHealthRecordRequest.getBlodtype());
+            log.info(errMessage);
+            throw new BadRequestException(errMessage);
+        }
+        RHFactor rhf = RHFactor.valueOfNotation(updateHealthRecordRequest.getRhfactor());
+        if (rhf == null) {
+            String errMessage = String.format("Nepoznat rh faktor '%s'", updateHealthRecordRequest.getRhfactor());
+            log.info(errMessage);
+            throw new BadRequestException(errMessage);
+        }
+
+        // getrecordbylbp
+        HealthRecord healthRecord = getRecordByLbp(lbp);
+
+        //update podatke
+        healthRecord = healthRecordMapper.updateHealthRecordRequestToHealthRecord(bt,rhf, healthRecord);
+
+        // update podatke u bazi
+        healthRecordRepository.save(healthRecord);
+
+        return new MessageResponse("uspesno azurirani podaci zdravstvenog kartona");
```

Comment:
Vrati ceo healthRecord (HealthRecordResponse ili LightHealthRecordResponse, mada je lakse sa LightHealthRecordResponse jer ima manje stvari, a i update-ujes samo rh i krvnu grupu, ne trebaju ti ostale stvari. Imas u mapperu vec funkciju, pa mozes nju da iskoristis. Mozes i da napravis novi response (npr. BasicHealthRecordResponse) koji ce da vrati samo id, rh, krvnu grupu i datum, da ne moras da se zajebavas sa vracanjem lista i sa njihovom paginacijom, pogotovo posto se nista od toga ne menja. Mozda je to najbolja opcija). Tako radimo da bi na frontu mogli odmah da prikazu sve promene, odnosno da prikazu stanje kartona kakvo je na beku, bez novog poziva bekenda.